test/ssl-tests/20-cert-select.cnf.in

```diff
@@ -914,7 +914,7 @@ my @tests_tls_1_3_non_fips = (
             #We only configured brainpoolP256r1 on the client side, but TLSv1.3
             #is enabled and this group is not allowed in TLSv1.3. Therefore this
             #should fail
-            "ExpectedResult" => "ServerFail"
+            "ExpectedResult" => "ClientFail"
```

Comment:
Why does this result change?

Reply:
This test should actually succeed, but unfortunatley I did not spot that earlier.
For reference #19325 just fixes the table in the provider that prevented the
test case to succeed in the first place,, Actually I found 3 more test cases,
which succeed once the above mentioned issues are resolved.

Reply:
Yes, I see, but my original intention was to keep this error prone distinction away from the API,
since there is already a well-known way to select the min and max TLS versions,
this creates just lots more ways to use the API in an inconsistent way.

Reply:
Unfortunately, IMO, having a single name for 2 groups ids creates more problems than it solves. It also breaks somewhat the provider capability mechanism where providers signal via capabilities which groups they support and tell libssl what the group ids and names it should be using are. Suddenly libssl needs to know about "special" groups that need to be handled differently to what the provider tells us about. In the event of more groups being added that work the same way as the brainpool groups then this adds additional confusion which will be dependent on libssl version rather than provider version. It actually has the potential to cause some chaos in the event that existing third party providers have already added support for the TLSv1.3 brainpool curves and are already using them in 3.0 (which is technically possible). Added to this the extra complexity in libssl is just not worth it. I spent most of a day playing whack-a-mole with bugs related to this complexity when trying to address an unrelated group code bug.

All of these problems disappear when we simply use the TLS capabilities mechanism in the way that it was originally designed (which is the approach taken in this PR).